polygeist/tools/cgeist/Lib/CodeGenTypes.cc

```diff
@@ -1583,43 +1580,194 @@ mlir::Type CodeGenTypes::getMLIRType(clang::QualType QT, bool *ImplicitRef,
+  case BuiltinType::Sampled##Id:
+#define IMAGE_WRITE_TYPE(Type, Id, Ext)
+#define IMAGE_READ_WRITE_TYPE(Type, Id, Ext)
+#include "clang/Basic/OpenCLImageTypes.def"
+#define EXT_OPAQUE_TYPE(ExtType, Id, Ext) case BuiltinType::Id:
+#include "clang/Basic/OpenCLExtensionTypes.def"
+  case BuiltinType::OCLSampler:
+  case BuiltinType::OCLEvent:
+  case BuiltinType::OCLClkEvent:
+  case BuiltinType::OCLQueue:
+  case BuiltinType::OCLReserveID:
+    return TypeTranslator.translateType(
+        CGM.getOpenCLRuntime().convertOpenCLSpecificType(BT));
+
+  case BuiltinType::SveInt8:
+  case BuiltinType::SveUint8:
+  case BuiltinType::SveInt8x2:
+  case BuiltinType::SveUint8x2:
+  case BuiltinType::SveInt8x3:
+  case BuiltinType::SveUint8x3:
+  case BuiltinType::SveInt8x4:
+  case BuiltinType::SveUint8x4:
+  case BuiltinType::SveInt16:
+  case BuiltinType::SveUint16:
+  case BuiltinType::SveInt16x2:
+  case BuiltinType::SveUint16x2:
+  case BuiltinType::SveInt16x3:
+  case BuiltinType::SveUint16x3:
+  case BuiltinType::SveInt16x4:
+  case BuiltinType::SveUint16x4:
+  case BuiltinType::SveInt32:
+  case BuiltinType::SveUint32:
+  case BuiltinType::SveInt32x2:
+  case BuiltinType::SveUint32x2:
+  case BuiltinType::SveInt32x3:
+  case BuiltinType::SveUint32x3:
+  case BuiltinType::SveInt32x4:
+  case BuiltinType::SveUint32x4:
+  case BuiltinType::SveInt64:
+  case BuiltinType::SveUint64:
+  case BuiltinType::SveInt64x2:
+  case BuiltinType::SveUint64x2:
+  case BuiltinType::SveInt64x3:
+  case BuiltinType::SveUint64x3:
+  case BuiltinType::SveInt64x4:
+  case BuiltinType::SveUint64x4:
+  case BuiltinType::SveBool:
+  case BuiltinType::SveFloat16:
+  case BuiltinType::SveFloat16x2:
+  case BuiltinType::SveFloat16x3:
+  case BuiltinType::SveFloat16x4:
+  case BuiltinType::SveFloat32:
+  case BuiltinType::SveFloat32x2:
+  case BuiltinType::SveFloat32x3:
+  case BuiltinType::SveFloat32x4:
+  case BuiltinType::SveFloat64:
+  case BuiltinType::SveFloat64x2:
+  case BuiltinType::SveFloat64x3:
+  case BuiltinType::SveFloat64x4:
+  case BuiltinType::SveBFloat16:
+  case BuiltinType::SveBFloat16x2:
+  case BuiltinType::SveBFloat16x3:
+  case BuiltinType::SveBFloat16x4: {
+    ASTContext::BuiltinVectorTypeInfo Info =
+        Context.getBuiltinVectorTypeInfo(BT);
+    llvm::Type *Ty = llvm::ScalableVectorType::get(
+        CGM.getTypes().ConvertType(Info.ElementType),
+        Info.EC.getKnownMinValue() * Info.NumVectors);
+    return TypeTranslator.translateType(Ty);
+  }
+
+#define PPC_VECTOR_TYPE(Name, Id, Size) case BuiltinType::Id:
+#include "clang/Basic/PPCTypes.def"
+#define RVV_TYPE(Name, Id, SingletonId) case BuiltinType::Id:
+#include "clang/Basic/RISCVVTypes.def"
+    llvm_unreachable("Unexpected PPC type");
+
+  case BuiltinType::Dependent:
+#define BUILTIN_TYPE(Id, SingletonId)
+#define PLACEHOLDER_TYPE(Id, SingletonId) case BuiltinType::Id:
+#include "clang/AST/BuiltinTypes.def"
+    llvm_unreachable("Unexpected placeholder builtin type!");
+  }
```

Comment:
should we add a llvm_unreachable after the switch to ensure no unexpected buildin type?

Reply:
Yes is a good idea.